packages/mnemonic-seed-manager/src/agent/MnemonicSeedManager.ts

```diff
@@ -113,4 +118,49 @@ export class MnemonicInfoGenerator implements IAgentPlugin {
       .orWhere('hash = :hash', { hash: args.hash })
       .execute()
   }
+
+  private async saveMasterKey(args: IMnemonicInfoStoreArgs): Promise<UpdateResult> {
+    if (args.masterKey) {
+      return (await this.dbConnection)
+        .createQueryBuilder()
+        .update(MnemonicInfo)
+        .set({ masterKey: args.masterKey })
+        .where('id = :id', { id: args.id })
+        .orWhere('hash = :hash', { hash: args.hash })
+        .execute()
+    }
+    throw new Error('Master Key needs to be provided.')
+  }
+
+  private async generateMasterKey(args: IMnemonicInfoStoreArgs): Promise<IMnemonicInfoKeyResult> {
+    const mnemonic = (await this.getMnemonicInfo({ id: args.id, hash: args.hash })).mnemonic
+    if (mnemonic) {
+      const seed = await bip39.mnemonicToSeed(mnemonic?.join(' '))
+      const { key, chainCode } = getMasterKeyFromSeed(seed.toString('hex'))
+      await this.saveMasterKey({ masterKey: key.toString('hex'), chainCode: chainCode.toString('hex') })
+      return { masterKey: key.toString('hex'), chainCode: chainCode.toString('hex') }
+    }
+    throw new Error('Mnemonic not found')
+  }
+
+  private async generateKeysFromMnemonic(args: IMnemonicInfoStoreArgs, context: IRequiredContext): Promise<ManagedKeyInfo> {
+    const mnemonic = (await this.getMnemonicInfo({ id: args.id, hash: args.hash })).mnemonic
+    if (mnemonic && context) {
+      if (args.path && args.kms) {
+        const seed = await bip39.mnemonicToSeed(mnemonic?.join(' '))
+        const { key, chainCode } = derivePath(args.path, seed.toString('hex'))
```

Comment:
You mentioned you fixed that we only can use hardned keys now in the derevation. Am I missing something as I cannot find the code for it.

Reply:
The default is hardened.

Reply:
But are we ensuring/enforcing that the user cannot use non-hardened keys?

Because that is what we need. We need to protect our users

Reply:
The method derivePath accepts a third optional argument called offset, which I'm not setting or allowing someone to set. The default offset is 0x80000000.